### ISSUE TITLE
listen udp4 0.0.0.0:123: bind: address already in use (only >= 3.2.0)

### ISSUE THREAD

**Issue by lucantolinos** [docker/for-mac#5589]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics

### Expected behavior
Container should start and port UDP 123 should be open.

### Actual behavior
Error starting userland proxy: listen udp4 0.0.0.0:123: bind: address already in use. The problem is only on the port 123, if I bind with the port 124, 80, 53 etc. no problem. I've checked with `sudo lsof -nP +c 15 | grep LISTEN` and `sudo nmap -sU localhost` that the port 123 is free on my host.

### Information
Reproductible YES.
the problem is appeared with version >= 3.2.0. 
Version 3.1.0 ==> OK
Version >= 3.2.0 ==> KO, address already in use.
No problem on Linux.

  - macOS Version: 10.15.7
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 3.3.1

### Steps to reproduce the behavior
Example with the hello-world container, juste trying to bind the udp 123 port from the container on my host : 

> docker run -p 123:123/udp --name hello hello-world
> docker: Error response from daemon: driver failed programming external connectivity on endpoint hello (467039b0bdf141aeaae62118ae38bbea809d0343e7e150089a579d3ac1f0f80b): Error starting userland proxy: listen udp4 0.0.0.0:123: bind: address already in use.
> ERRO[0002] error waiting for container: context canceled

Example with a real NTP image that open bind a NTP server on the port 123 :
> docker run  --publish=123:123/udp    cturra/ntp

**Comment by cturra** [comment]
thanks for reporting this @lucantolinos. i wanted to jump on this issue and share that i am experiencing the same behaviour as you're describing (and am the author of the [cturra/ntp](https://hub.docker.com/r/cturra/ntp/) container).

similar to this report, i get the same results with both docker desktop on mac 3.3.1 and 3.3.0. here are the steps to reproduce from my environment (macos 11.2.3; Intel).

fetch container:
```
$ docker pull cturra/ntp
Using default tag: latest
latest: Pulling from cturra/ntp
Digest: sha256:138d7cace3bab5113a5944fdae10e91e406bfa0f0fe013afd908eee09839a147
Status: Image is up to date for cturra/ntp:latest
docker.io/cturra/ntp:latest
```

launch:
```
$ docker run --name=ntp --detach --publish=123:123/udp cturra/ntp
efdd358c3e4b6f11e7ff269afac3104a61512d2a28caf0a3b52307a5050015c8
docker: Error response from daemon: driver failed programming external connectivity on endpoint ntp (c82dc97659eb46289f368d20e89a094ad96bb429f770762e79a31f221282c168): Error starting userland proxy: listen udp4 0.0.0.0:123: bind: address already in use.
```

confirm port not in use:
```
$ sudo lsof -i :123 | echo $?
1

$ netstat -anp udp | awk '{print $4}' | grep 123 | echo $?
1
```

thanks folks! and please, let us know if there's anything more we can help with to get this resolved.

**Comment by stephen-turner** [comment]
Thanks for the report. We had a similar problem with port 53 (DNS) earlier, which we fixed, but we didn't notice that port 123 also had the same problem. We'll fix it.

**Comment by filoucrackeur** [comment]
same problem using ipa server with dns  on port 123 freeipa/freeipa-server:fedora-rawhide-4.9.3

**Comment by mat007** [comment]
We’ve released 3.3.2 that fixes this issue, see [the release notes](https://docs.docker.com/docker-for-mac/release-notes/).
Thanks!

**Comment by cturra** [comment]
confirmed, this now works as expected! thanks to everyone involved in helping get this resolved. ❤️
